sample-apps/discount-functions-sample-app/extensions/fixed-amount-product-discount/src/main.rs

```diff
@@ -0,0 +1,229 @@
+            Some(input::Metafield { value }) => Configuration::from_str(value),
+            None => Configuration::default(),
+        }
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let input: input::Input = serde_json::from_reader(std::io::BufReader::new(std::io::stdin()))?;
+    let mut out = std::io::stdout();
+    let mut serializer = serde_json::Serializer::new(&mut out);
+    function(input)?.serialize(&mut serializer)?;
+    Ok(())
+}
+
+fn function(input: input::Input) -> Result<FunctionResult, Box<dyn std::error::Error>> {
+    let cart_lines = &input.cart.lines;
+    let config: Configuration = input.configuration();
+
+    let converted_value = convert_to_cart_currency(config.value, &input);
+    let targets = targets(&cart_lines);
+    Ok(build_result(converted_value, targets))
+}
+
+fn convert_to_cart_currency(value: f64, input: &input::Input) -> f64 {
+    let rate = &input.presentment_currency_rate;
+    value * rate.parse::<f64>().expect("presentment_currency_rate is malformed.")
+}
+
+fn targets(cart_lines: &[input::CartLine]) -> Vec<Target> {
+    cart_lines
+        .iter()
+        .filter_map(|line| match &line.merchandise.id {
+            Some(id) => {
+                Some(Target::ProductVariant {
+                    id: id.to_string(),
+                    quantity: None,
+                })
+            }
+            None => None
+        })
+        .collect()
+}
+
+fn build_result(amount: f64, targets: Vec<Target>) -> FunctionResult {
+    let discounts = if targets.is_empty() {
+        vec![]
+    } else {
+        vec![Discount {
+            message: None,
+            conditions: None,
+            targets,
+            value: Value::FixedAmount(FixedAmount {
+                amount: format!("{}", amount),
+                applies_to_each_item: None
+            }),
+        }]
+    };
+    FunctionResult {
+        discounts,
+        discount_application_strategy: DiscountApplicationStrategy::First,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn input(configuration: Option<Configuration>, presentment_currency_rate: Option<Decimal>) -> input::Input {
+        let input = json!(
+        {
+            "cart": {
+                "lines": [
+                    {
+                        "id": "gid://shopify/CartLine/0",
+                        "merchandise": {
+                            "id": "gid://shopify/ProductVariant/0"
+                        }
+                    },
+                    {
+                        "id": "gid://shopify/CartLine/1",
+                        "merchandise": {
+                            "id": "gid://shopify/ProductVariant/1"
+                        }
+                    }
+                ]
+            },
+            "discountNode": { "metafield": null },
+            "presentmentCurrencyRate": "1.00"
+        });
+        let default_input: input::Input = serde_json::from_value(input).unwrap();
```

Comment:
Why go through `serde_json::Value` and not just construct an `input::Input`? That way any errors occur at compile-time (and can be detected by `rust-analyzer`) instead of during the tests